### ISSUE TITLE
R-4.2.2-foss-2022b error at extension lpSolve 

### ISSUE THREAD

**Issue by DirectorSloan** [easybuilders/easybuild-easyconfigs#17708]
When trying to install `R-4.2.2-foss-2022b` it fails at extension `lpSolve` (5.6.18) it breaks with complaining "fPIC command not found". I've quickly tried it with a locally R (4.2.1) and manual `R CMD INSTALL lpSolve_5.6.18.tar.gz`  works fine.

Maybe something in connecting the buildopts together here.

Here my `eb --show-config`

```
admbela@frontend2 08:28 /software/easybuild-share/easybuild-easyconfigs/easybuild/easyconfigs $ eb --show-config
#
# Current EasyBuild configuration
# (C: command line argument, D: default value, E: environment variable, F: configuration file)
#
buildpath                 (E) = /tmp/easybuild-build
containerpath             (E) = /software/easybuild-staging/container
cuda-compute-capabilities (E) = 3.7, 7.0, 8.0
default-opt-level         (E) = opt
enforce-checksums         (E) = True
include-easyblocks        (E) = /software/easybuild-share/easyblocks/*.py
installpath               (E) = /software/easybuild-staging
minimal-toolchains        (E) = True
module-depends-on         (E) = True
module-naming-scheme      (E) = HierarchicalMNS
optarch                   (E) = march=broadwell
packagepath               (E) = /software/easybuild-staging/packages
parallel                  (E) = 10
prefix                    (E) = /software/easybuild-staging
repositorypath            (E) = /software/easybuild-staging/ebfiles_repo
robot-paths               (E) = /software/easybuild-share/easybuild-easyconfigs/easybuild/easyconfigs
sourcepath                (E) = /software/easybuild-share/sources
umask                     (E) = 002
zip-logs                  (E) = gzip

``` 
I will attach my build log.

**Comment by branfosj** [comment]
That is one odd error. There should be a `gcc` there. My initial guess is that some variable has a trailing newline character and that is causing a command to get split. I'm not sure I have suggestions on how to debug that though.

**Comment by DirectorSloan** [comment]
Yes, so that the dash is vanished and he interprets `fPIC` as command. May I reproduce it in another lab.

**Comment by branfosj** [comment]
`lpSolve` is early in the extension list, so my steps to debugging would be (this assumes you are using a test environment):
1. Create a new R easyconfig, based on the 4.2.2 one, that deletes all extensions after `lpSolve`
2. Install that with `--skip-extensions`
3. Then only install the extensions with `--skip --rebuild`

Then
1. Add a `'preinstallopts': "env && ",` to the `lpSolve` block and see if there are any odd line splits
2. Look in the build directory and see if there are any makefiles, etc., with a line split issue

**Comment by DirectorSloan** [comment]
Thanks a lot @branfosj, I'll try this tomorrow

**Comment by DirectorSloan** [comment]
> Thanks a lot @branfosj, I'll try this tomorrow

Managed to follow the plan but can't find any line splits that aare odd.

**Comment by DirectorSloan** [comment]
I've played around a little bit more and try to adapt the R-4.1.2-foss-2021b ebuild into the foss-2022b toolchain in order to find out if its the R version, the toolchain or something else because the last successfull R build was R 4.0.4 within 2020b.

And in this run the extension `Rmpi` breaks with the same kind of error (-fPIC expected in line beginning). Maybe the Python version when using EasyBuild 4.7.1. ...

Keep on trying...

**Comment by DirectorSloan** [comment]
https://stackoverflow.com/questions/50220186/r-package-source-install-no-compiler brought me to the correct idea.

Local `Makevars` in my `.R` directory destroyed the build behaviour:

```
CC = $(which gcc) -fPIC
CXX17 = $(which g++) -fPIC
CXX17STD = -std=c++17
CXX17FLAGS = ${CXX11FLAGS}
```

Needed this for another R build.
Drives me hell. Sorry for thinking loud here and bothering you with this.
